src/ast/ast_validations.c

```diff
@@ -883,24 +873,6 @@ static AST_Validation _Validate_UNWIND_Clauses(const AST *ast, char **reason) {
 	uint clause_count = array_len(unwind_clauses);
 	for(uint i = 0; i < clause_count; i++) {
 		const cypher_astnode_t *expression = cypher_ast_unwind_get_expression(unwind_clauses[i]);
-		cypher_astnode_type_t type = cypher_astnode_type(expression);
-		// Ensure that the UNWIND expression is a collection (aka list/array)
-		if(type == CYPHER_AST_APPLY_OPERATOR) {
-			const cypher_astnode_t *funcNode =  cypher_ast_apply_operator_get_func_name(expression);
-			const char *funcName = cypher_ast_function_name_get_value(funcNode);
-			// function name is NOT range
-			if(strcasecmp(funcName, "range")) {
-				asprintf(reason, "UNWIND expects range function; encountered '%s'", funcName);
-				res = AST_INVALID;
-				goto cleanup;
-			}
-		} else if(!(type == CYPHER_AST_COLLECTION || type == CYPHER_AST_IDENTIFIER ||
-					type == CYPHER_AST_PROPERTY_OPERATOR)) {
-			asprintf(reason, "UNWIND expects a list argument or an list identifier; encountered ''%s'",
-					 cypher_astnode_typestr(type));
-			res = AST_INVALID;
-			goto cleanup;
-		}
```

Comment:
Ideally we'd be able to capture things like `UNWIND 5 AS a RETURN a` as syntax errors, though I'd be fine with a TODO for the moment.

We should add some run-time captures in `UnwindConsume`, however. While this example just returns an empty result set, `UNWIND toString(5) AS a RETURN a` causes a crash for me.

Reply:
👍 Fixed